### PR TITLE
Dockerfile: Added dev environment dockerfile

### PR DESCRIPTION
**Pull request by johnandersen777**
Dockerfile builds a container containing all Zephyr's build deps and deps
for the net-tools repo. This enables developers to get started hacking quickly
or in the case of QA to verify bugs from a clean installation.

Signed-off-by: John Andersen <john.s.andersen@intel.com>

### CODE REVIEW

**Review by nashif — CHANGES_REQUESTED** [review]
put all the files under scripts/docker or something like that.

**Comment by pdxjohnny** [comment]
@nashif Put them under `scripts/`

File: scripts/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
+FROM ubuntu:16.04
+ARG no_proxy
+ARG http_proxy
+ARG https_proxy
+ENV no_proxy ${no_proxy}
+ENV http_proxy ${http_proxy}
+ENV https_proxy ${https_proxy}
+
+ARG CMAKE_URL
+ENV CMAKE_URL ${CMAKE_URL:-https://cmake.org/files/v3.8/cmake-3.8.2-Linux-x86_64.sh}
```
Comment (nashif) [inline scripts/docker/Dockerfile]: in CI we are using cmake 3.9, we need to think about how we can keep this in sync with our CI images hosted at https://github.com/zephyrproject-rtos/ci-dockerfiles

File: scripts/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
+FROM ubuntu:16.04
```
Comment (nashif) [inline scripts/docker/Dockerfile]: need header, copyright, etc.

**Review by nashif — CHANGES_REQUESTED** [review]
Can you add some documentation on how to use this? We also need to think about how to make this compatible with CI docker image of Zephyr, maybe share some of the install steps instead of maintaining them separately.

**Comment by galak** [comment]
Any plans to address @nashif comments or should we close this?

**Comment by galak** [comment]
Closing this as no comments after a week to update.

**Pull request closed without merging**